### PR TITLE
Add tabs and post tweet feature

### PR DESCRIPTION
**Pull request by Vadifire**

### CODE REVIEW

File: src/main/js/components/PostTweetUI.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+					<button className="post-tweet" onClick={this.postTweet} 
+								disabled={!(this.state.input.replace(/\s/g, "").length)}>
+						{PostTweetUI.buttonText}
+					</button>
+				</div>
+			</div>
+		);
+	}
+
+	updateMessage(event) {
```
Comment (briansheen) [inline src/main/js/components/PostTweetUI.js]: can you move methods above the render?

File: src/main/js/components/PostTweetUI.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		this.setState({input: event.target.value});
+	}
+
+	postTweet() {
+		postTweet((err) => {
```
Comment (briansheen) [inline src/main/js/components/PostTweetUI.js]: looks like you are calling a function to call another function?  is there a way to refactor this?

File: src/main/js/components/PostTweetUI.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	static get failureMessage() {
+		return "Could not post tweet.";
+	}
+
+	updateMessage(event) {
+		this.setState({input: event.target.value});
+	}
+
+	submitTweet() {
+		postTweet((err) => {
```
Comment (briansheen) [inline src/main/js/components/PostTweetUI.js]: if you're worried about readability, is there a way to not pass in the callback inline?

File: src/main/js/components/PostTweetUI.js [hunk 4]
```diff
@@ -53,7 +54,7 @@ export default class PostTweetUI extends React.Component {
 						{this.state.output}
 					</div>
 					<button className="post-tweet"
-							onClick={() => this.postTweet(this.state.input)}
+							onClick={() => postTweet(this.postCallback, this.state.input)}
```
Comment (briansheen) [inline src/main/js/components/PostTweetUI.js]: nice

File: src/main/css/components/PostTweetUI.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	
+	.input-container {
+		width: 100%;
+		position: relative;
+
+		.char-count {
+			position: absolute;
+			float: right;
+			z-index: 5; // Put in front of tweet-input
+			pointer-events: none;
```
Comment (kev-chan) [inline src/main/css/components/PostTweetUI.scss]: what does this do?

File: src/main/css/components/TabContainer.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	display: flex;
+	justify-content: center;
+}
+
+.tabs {
+	text-align: center;
+	.nav-tabs {
+		@include center;
+		list-style-type: none;
+		padding: 0px; // Removes padding from invisible bullets
```
Comment (kev-chan) [inline src/main/css/components/TabContainer.scss]: just say 0

File: src/main/css/components/TabContainer.scss [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+		display: inline-block;
+
+		.nav-item {
+			display: flex;
+			margin: 0px 2px;
```
Comment (kev-chan) [inline src/main/css/components/TabContainer.scss]: same as above

**Pull request merged**